includes/class-wc-stripe-payment-method-configurations.php

```diff
@@ -129,13 +129,150 @@ private static function get_payment_method_configuration_from_stripe() {
+	 * }
+	 */
+	private static function get_fallback_payment_method_configuration( array $payment_method_configurations ): array {
+		$active_non_child_payment_method_configurations = array_filter(
+			$payment_method_configurations,
+			function ( $configuration ) {
+				if ( $configuration->parent ?? null ) {
+					return false;
+				}
+				return isset( $configuration->active ) && $configuration->active;
+			}
+		);
+
+		$fallback_pmc_id = get_option( 'woocommerce_stripe_pmc_fallback_id' );
+		$fallback_pmc    = null;
+
+		if ( [] === $active_non_child_payment_method_configurations ) {
+			if ( $fallback_pmc_id ) {
+				WC_Stripe_Logger::debug(
+					'No Payment Method Configurations returned from Stripe; deleting fallback ID option',
+					[
+						'fallback_pmc_id' => $fallback_pmc_id,
+					]
+				);
+				delete_option( 'woocommerce_stripe_pmc_fallback_id' );
+			}
+
+			return [
+				'pmc'     => null,
+				'reason'  => 'no_eligible_pmcs',
+			];
+		}
+
+		if ( $fallback_pmc_id ) {
+			foreach ( $active_non_child_payment_method_configurations as $configuration ) {
+				if ( $configuration->id === $fallback_pmc_id ) {
+					return [
+						'pmc'    => $configuration,
+						'reason' => 'existing_fallback_pmc_used',
+					];
+				}
+			}
+
+			// If we get here and don't have our fallback yet, we need to delete the fallback ID option.
+			WC_Stripe_Logger::debug(
+				'Fallback Payment Method Configuration not returned from Stripe, deleting fallback ID option',
+				[
+					'fallback_pmc_id' => $fallback_pmc_id,
+				]
+			);
+			delete_option( 'woocommerce_stripe_pmc_fallback_id' );
+			$fallback_pmc_id = null;
+		}
+
+		$fallback_reason = null;
+		// If we only have one remaining PMC, it should be the fallback, so we don't need to check for default PMCs.
+		if ( 1 === count( $active_non_child_payment_method_configurations ) ) {
+			$fallback_pmc    = reset( $active_non_child_payment_method_configurations );
+			$fallback_reason = 'only_pmc_used';
+		}
+
+		if ( ! $fallback_pmc ) {
+			$default_payment_method_configurations = array_filter(
+				$active_non_child_payment_method_configurations,
+				function ( $configuration ) {
+					return isset( $configuration->default ) && $configuration->default;
+				}
+			);
+
+			if ( 1 === count( $default_payment_method_configurations ) ) {
+				// Use reset() as array_filter() preserves keys.
+				$fallback_pmc    = reset( $default_payment_method_configurations );
+				$fallback_reason = 'only_default_pmc_used';
+			}
+		}
+
+		if ( $fallback_pmc ) {
+			WC_Stripe_Logger::debug(
+				'Updating Payment Method Configuration fallback',
+				[
+					'pmc_id'   => $fallback_pmc->id,
+					'reason'   => $fallback_reason,
+					'name'     => $fallback_pmc->name ?? null,
+					'livemode' => $fallback_pmc->livemode ?? null,
+				]
+			);
+			update_option( 'woocommerce_stripe_pmc_fallback_id', $fallback_pmc->id );
```

Comment:
nit: Should we store this under separate keys for live and test mode? Although, given this is a last-resort path, the value added might be low relative to the complexity introduced.

Reply:
I've split the option keys in b54d1d421a3bc72c3d72ce288ccc09ef9d6fc02a. I think the use of dedicated options will be more robust in the long term, as the intent and expectations are explicit.